core/services/keystore/keys/tronkey/account.go

```diff
@@ -0,0 +1,175 @@
+package tronkey
+
+import (
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/mr-tron/base58"
+)
+
+// Extracted from go-tron sdk: https://github.com/fbsobreira/gotron-sdk
+
+const (
+	// HashLength is the expected length of the hash
+	HashLength = 32
+	// AddressLengthBase58 is the expected length of the address in base58format
+	AddressLengthBase58 = 34
+	// Tron Address Prefix
+	prefixMainnet = 0x41
+	// TronBytePrefix is the hex prefix to address
+	TronBytePrefix = byte(prefixMainnet)
+	// Tron address should should have 20 bytes + 4 checksum + 1 Prefix
+	AddressLength = 20
+)
+
+// Address represents the 21 byte address of an Tron account.
+type Address []byte
+
+// Bytes get bytes from address
+func (a Address) Bytes() []byte {
+	return a[:]
+}
+
+// Hex get bytes from address in string
+func (a Address) Hex() string {
+	return BytesToHexString(a[:])
+}
+
+// HexToAddress returns Address with byte values of s.
+// If s is larger than len(h), s will be cropped from the left.
```

Comment:
I don't think this is true currently
```suggestion
// If s is larger than len(h), s will be cropped from the left.
```

Reply:
changed the implementation of this slightly, although not sure if its actually used at the moment